utils/UUID.hh

```diff
@@ -161,7 +162,7 @@ inline uint64_t timeuuid_read_msb(const int8_t *b) {
            u64(b[2]) << 8  | u64(b[3]);
 }
 
-inline uint64_t uuid_read_lsb(const int8_t *b) {
+inline uint64_t uuid_read_lsb(managed_bytes_view b) {
```

Comment:
I'm not sure if it's the right thing to make `utils/UUID.hh` dependent on `managed_bytes`.

UUIDs have fixed size, it's safe to linearize them without causing a big allocation. I would rather leave `utils/UUID.hh` unchanged and just call `with_linearized()` in `compare_visitor`. Especially since `operator[]` for `managed_bytes_view` is inefficient.